### PR TITLE
Add a CONTEXT fragment to the readme generator

### LINKED ISSUE #571: Add a new CONTEXT section into README.rst
As part of the project to include / attract more functional people into contributing to OCA modules, there have been discussions around the use of the Read me and how functionals could be contributing at modules documentation.

One of the ideas / needs for functional people is to be able to document the context, some use cases, the why of the module.

So, we came with the idea of adding a new file/section to the Read Me dedicated to this topic.

It could be named : **Context/Use cases**

It could be added after the _Description_ but before the _Installation_.
Proposal_add_section_read_me

### PR DESCRIPTION
**Pull request by sbidoul**
closes #571

### CODE REVIEW

**Comment by sbidoul** [comment]
The name of the `CONTEXT` fragment and the title in the generated README.rst (currently `When to use`) are up for bike shedding.

**Comment by TumbaoJu** [comment]
@sbidoul 

Here is the proposition of the Functional Group.

**Name :** Context
**Title :** Use Cases / Context

**Description :** 
[This file is optional but strongly suggested to allow end-users to evaluate the module’s usefulness in their context. 

It should explain the “why” of the module: 
- in which context or use cases this module can be useful (practical examples are welcome!). 
- what is the business requirement that generated the need to develop this module

It can also inform on related modules:
- modules it depends on and their features
- other modules that can work well together with this one 
- suggested setups where the module is useful (eg: multicompany, multi-website) ]

**Review by francesco-ooops — APPROVED** [review]

**Review by rvalyi — APPROVED** [review]
LGTM. We used to do it in the module description, but having it structured is probably better indeed.

File: template/module/readme/CONTEXT.rst [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[ This file is optional but strongly suggested to allow end-users to evaluate the
+module's usefulness in their context. ]
```
Comment (yajo) [inline template/module/readme/CONTEXT.rst:2]:
```suggestion
module's usefulness in their context.
```

The closing bracket seems to be at the end of the file.

**Review by yajo — CHANGES_REQUESTED** [review]

File: template/module/readme/CONTEXT.rst (hunk 1 above)
Comment (TumbaoJu) [reply template/module/readme/CONTEXT.rst:2]: If we look at other files, we can see that the bracket is at the end of the main sentence.
description_fragment_text_description

**Review by TumbaoJu — APPROVED** [review]

**Review by TumbaoJu — APPROVED** [review]

File: template/module/readme/CONTEXT.rst (hunk 1 above)
Comment (yajo) [reply template/module/readme/CONTEXT.rst:2]: OK, then...

File: template/module/readme/CONTEXT.rst [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+It should explain the “why” of the module:
+
+- in which context or use cases this module can be useful (practical examples are welcome!).
+- what is the business requirement that generated the need to develop this module
+
+It can also inform on related modules:
+
+- modules it depends on and their features
+- other modules that can work well together with this one
+- suggested setups where the module is useful (eg: multicompany, multi-website) ]
```
Comment (yajo) [inline template/module/readme/CONTEXT.rst:13]: ... remove this one.

```suggestion
- suggested setups where the module is useful (eg: multicompany, multi-website)
```

**Review by yajo — CHANGES_REQUESTED** [review]

**Review by TumbaoJu — APPROVED** [review]

**Pull request merged**